### PR TITLE
Adding prow-arm-private config.

### PR DESCRIPTION
**Pull request by stewartbutler**

### CODE REVIEW

File: prow/cluster/private-arm/prow-private-istio-io_managedcertificate.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: networking.gke.io/v1
+kind: ManagedCertificate
```
Comment (howardjohn) [inline prow/cluster/private-arm/prow-private-istio-io_managedcertificate.yaml]: we shouldn't need this

File: prow/cluster/private-arm/kubernetes_external_secrets.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Empty
```
Comment (howardjohn) [inline prow/cluster/private-arm/kubernetes_external_secrets.yaml:1]: nit: if its empty do we really need prow/cluster/private-arm/kubernetes-external-secrets_deployment.yaml etc?

File: prow/cluster/private-arm/release-deps.yaml [hunk 3]
```diff
@@ -0,0 +1,257 @@
+# Below contains the configmaps that configure the release-builder manifests in istio/istio and istio/release-builder.
+# This should be updated for each version
+apiVersion: v1
```
Comment (howardjohn) [inline prow/cluster/private-arm/release-deps.yaml]: not needed, we don't run this job on arm and will make burden to sync it

File: prow/config.yaml [hunk 4]
```diff
@@ -41,6 +41,10 @@ plank:
     config:
       default_service_account_name: "prowjob-default-sa" # Use workload identity
       gcs_credentials_secret: ""                         # rather than service account key secret
+  - cluster: prow-arm-private
+    config:
+      default_service_account_name: "prowjob-default-sa" # Use workload identity
+      gcs_credentials_secret: ""                         # rather than service account key secret
```
Comment (howardjohn) [inline prow/config.yaml]: should be like the istio-private one

```
      default_service_account_name: "prowjob-private-sa"
      gcs_configuration:
        bucket: "istio-private-build"
        path_strategy: "explicit"
      ssh_key_secrets:
      - ssh-key-secret
```

Comment (stewartbutler) [reply prow/config.yaml]: So the thing that confuses me is that the istio-private one does not specify a cluster, it specifies a 'repo'. What isn't clear is if that is decorating something else, or it is still applying to a cluster.

File: prow/cluster/private-arm/kubernetes_external_secrets.yaml (hunk 2 above)
Comment (stewartbutler) [reply prow/cluster/private-arm/kubernetes_external_secrets.yaml:1]: So I'm copying the convention from `arm`. Should I likewise delete that?

File: prow/cluster/private-arm/prow-private-istio-io_managedcertificate.yaml (hunk 1 above)
Comment (stewartbutler) [reply prow/cluster/private-arm/prow-private-istio-io_managedcertificate.yaml]: Nuked.

File: prow/cluster/private-arm/release-deps.yaml (hunk 3 above)
Comment (stewartbutler) [reply prow/cluster/private-arm/release-deps.yaml]: Nuked.

**Review by jacob-delgado — APPROVED** [review]

File: prow/config.yaml (hunk 4 above)
Comment (howardjohn) [reply prow/config.yaml]: weird, need to look into the docs. Regardless prowjob-default-sa seems wrong , seems like we have a private SA?

Comment (howardjohn) [reply prow/config.yaml]: you can match org or repo (confusing key, it's actually matching the org). https://github.com/kubernetes/test-infra/blob/c66af3fdd561/prow/config/config.go#L721

so we. probably want to remove this entirely

File: prow/cluster/crier_deployment.yaml [hunk 5]
```diff
@@ -35,7 +35,7 @@ spec:
         env:
         # Use KUBECONFIG envvar rather than --kubeconfig flag in order to provide multiple configs to merge.
         - name: KUBECONFIG
-          value: "/etc/kubeconfig/istio-config:/etc/kubeconfig-build-test-infra-trusted/kubeconfig"
+          value: "/etc/kubeconfig/istio-config:/etc/kubeconfig-build-test-infra-trusted/kubeconfig:/etc/kubeconfig-prow-arm-private/kubeconfig"
```
Comment (howardjohn) [inline prow/cluster/crier_deployment.yaml:38]: I don't think we want private here, based on the other private note being there.

File: prow/cluster/private-arm/kubernetes_external_secrets.yaml (hunk 2 above)
Comment (howardjohn) [reply prow/cluster/private-arm/kubernetes_external_secrets.yaml:1]: I guess so, maybe we do it later though

File: prow/cluster/crier_deployment.yaml (hunk 5 above)
Comment (howardjohn) [reply prow/cluster/crier_deployment.yaml:38]: Oh the `istio-config` has all of them today, I see

**Review by howardjohn — APPROVED** [review]

**Pull request merged**